### PR TITLE
Sonde YAML with QCs

### PR DESCRIPTION
**Pull request by PraveenKumar-NOAA**
A new YAML is written for the sonde data, which performs better than the old one for the GSI/JEDI QC obs, and added into the config and testing directories of the GDASapp.

The Sonde Obs Validation check is performed for the followings: 

- HofX computation using GSI produced GeoVaLs
- QC count checks (and YAML changes) using GSI produced GeoVaLs
- Obs errors using GSI-produced GeoVaLs

The results were produced for the following observations: 
surface_pressure(#41: https://github.com/NOAA-EMC/JEDI-T2O/issues/41)
air_temperature(#41: https://github.com/NOAA-EMC/JEDI-T2O/issues/41)
eastward_wind(#50: https://github.com/NOAA-EMC/JEDI-T2O/issues/50)
northward-wind(#50: https://github.com/NOAA-EMC/JEDI-T2O/issues/50)
specific_humidity(#51: https://github.com/NOAA-EMC/JEDI-T2O/issues/51).

### CODE REVIEW

File: parm/atm/obs/config/sondes.yaml [hunk 1]
```diff
@@ -1,74 +1,44 @@
...
       type: H5File
-      obsfile: $(DIAG_DIR)/diag_sondes_$(OBS_DATE).nc4
-  io pool:
-    max pool size: 1
-  simulated variables: [eastward_wind, northward_wind, air_temperature, specific_humidity, surface_pressure]
-get values:
-  interpolation method: $(INTERP_METHOD)
+      obsfile: sondes_diag_$(OBS_DATE).nc4
+  simulated variables: [surface_pressure, air_temperature, eastward_wind, northward_wind, specific_humidity]
+geovals:
```
Comment (CoryMartin-NOAA) [inline parm/atm/obs/config/sondes.yaml]: geovals, vector ref, tolerance do not go in the `config` YAMLs. Also the `io pool` section needs to stay

File: parm/atm/obs/config/sondes.yaml [hunk 2]
```diff
@@ -1,74 +1,44 @@
...
-     - name: northward_wind
-   - name: Identity
-     variables:
+     geovar_sfc_geomz: surface_geometric_height
+     geovar_geomz: geopotential_height
+     variables: 
      - name: surface_pressure
-#
+linear obs operator:
+  name: Identity
```
Comment (CoryMartin-NOAA) [inline parm/atm/obs/config/sondes.yaml]: I don't think linear obs operator should be `identity` for all ob types here

File: parm/atm/obs/config/sondes.yaml [hunk 3]
```diff
@@ -1,74 +1,44 @@
...
-  #   action:
-  #     name: reject
-  #
-  # Observation Range Sanity Check
-  - filter: Bounds Check
-    filter variables:
-    - name: air_temperature
-    minvalue: 195
-    maxvalue: 327
+  - filter: PreQC
```
Comment (CoryMartin-NOAA) [inline parm/atm/obs/config/sondes.yaml:50]: Is there a reason why there is like almost no QC retained in this file?

File: parm/atm/obs/config/sondes.yaml [hunk 4]
```diff
@@ -1,74 +1,44 @@
...
   name: sondes
   obsdatain:
-    engine:
+    engine: 
       type: H5File
-      obsfile: $(OBS_DIR)/$(OBS_PREFIX)sondes.$(OBS_DATE).nc4
-    obsgrouping:
-      group variables: ["station_id"]
-      sort variable: "air_pressure"
-      sort order: "descending"
```
Comment (nicholasesposito) [inline parm/atm/obs/config/sondes.yaml]: More of a general question, but why is this part removed?

File: parm/atm/obs/config/sondes.yaml (hunk 1 above)
Comment (PraveenKumar-NOAA) [reply parm/atm/obs/config/sondes.yaml]: Updated for the required changes.

File: parm/atm/obs/config/sondes.yaml (hunk 2 above)
Comment (PraveenKumar-NOAA) [reply parm/atm/obs/config/sondes.yaml]: Updated linear obs operator for all the obs variables used in the file.

File: parm/atm/obs/config/sondes.yaml (hunk 4 above)
Comment (PraveenKumar-NOAA) [reply parm/atm/obs/config/sondes.yaml]: I don't think it was necessary, but I added this section back again to be consistent with the old YAML.

File: parm/atm/obs/config/sondes.yaml (hunk 3 above)
Comment (PraveenKumar-NOAA) [reply parm/atm/obs/config/sondes.yaml:50]: I mostly removed all the QC for all the variables, except surface_pressure, and applied the PreQC filter. It worked better compared to the old YAML. For a comparison of the number of obs passing QCs with different YAMLs (no QC, old YAML, and new YAML), please look at the following document: https://docs.google.com/document/d/1_w6P-cJkxyYzTvLNMm6kCBWj2ZATk0PMsXOdKW29z8s/edit.

Comment (PraveenKumar-NOAA) [reply parm/atm/obs/config/sondes.yaml:50]: Please look at my sonde obs yaml PR.

File: parm/atm/obs/config/sondes.yaml (hunk 4 above)
Comment (PraveenKumar-NOAA) [reply parm/atm/obs/config/sondes.yaml]: @nicholasesposito please look at this PR for sonde obs YAML.

File: parm/atm/obs/config/sondes.yaml [hunk 5]
```diff
@@ -1,74 +1,56 @@
...
+      obsfile: sondes_obs_$(OBS_DATE).nc4
+      obsgrouping:
```
Comment (CoryMartin-NOAA) [inline parm/atm/obs/config/sondes.yaml]: I think this indentation is now wrong, it should be unchanged I think.

File: parm/atm/obs/config/sondes.yaml [hunk 6]
```diff
@@ -1,74 +1,56 @@
...
   obsdatain:
-    engine:
+    engine: 
       type: H5File
-      obsfile: $(OBS_DIR)/$(OBS_PREFIX)sondes.$(OBS_DATE).nc4
-    obsgrouping:
-      group variables: ["station_id"]
-      sort variable: "air_pressure"
-      sort order: "descending"
+      obsfile: sondes_obs_$(OBS_DATE).nc4
```
Comment (CoryMartin-NOAA) [inline parm/atm/obs/config/sondes.yaml]: obsfile should not be changed, it should be the same as before

File: parm/atm/obs/config/sondes.yaml [hunk 7]
```diff
@@ -1,74 +1,56 @@
...
+        sort order: "descending"
   obsdataout:
     engine:
       type: H5File
-      obsfile: $(DIAG_DIR)/diag_sondes_$(OBS_DATE).nc4
+      obsfile: sondes_diag_$(OBS_DATE).nc4
   io pool:
     max pool size: 1
-  simulated variables: [eastward_wind, northward_wind, air_temperature, specific_humidity, surface_pressure]
-get values:
```
Comment (CoryMartin-NOAA) [inline parm/atm/obs/config/sondes.yaml:18]: why are `get values` and `interpolation method` removed?

File: parm/atm/obs/testing/sondes.yaml [hunk 8]
```diff
@@ -0,0 +1,106 @@
+obs space:
+  name: sondes
+  obsdatain:
+    engine: 
+      type: H5File
+      obsfile: sondes_obs_$(OBS_DATE).nc4
+      obsgrouping:
```
Comment (CoryMartin-NOAA) [inline parm/atm/obs/testing/sondes.yaml]: same question, is this the right indentation for obsgrouping? (not 100% sure)

File: parm/atm/obs/config/sondes.yaml [hunk 9]
```diff
@@ -184,150 +91,14 @@ obs filters:
         options:
           error_min: 100         # 1 mb
```
Comment (CoryMartin-NOAA) [inline parm/atm/obs/config/sondes.yaml]: Based on our Wednesday group discussion, I think we need to make clear these are 'floats' and not integers (I think there are a few spots throughout this YAML).

File: parm/atm/obs/config/sondes.yaml (hunk 5 above)
Comment (PraveenKumar-NOAA) [reply parm/atm/obs/config/sondes.yaml]: @CoryMartin-NOAA fixed.

File: parm/atm/obs/config/sondes.yaml (hunk 6 above)
Comment (PraveenKumar-NOAA) [reply parm/atm/obs/config/sondes.yaml]: @CoryMartin-NOAA fixed.

File: parm/atm/obs/config/sondes.yaml (hunk 7 above)
Comment (PraveenKumar-NOAA) [reply parm/atm/obs/config/sondes.yaml:18]: @CoryMartin-NOAA fixed.

File: parm/atm/obs/testing/sondes.yaml (hunk 8 above)
Comment (PraveenKumar-NOAA) [reply parm/atm/obs/testing/sondes.yaml]: @CoryMartin-NOAA I think so. A similar indentation is used in other YAML.

File: parm/atm/obs/config/sondes.yaml (hunk 9 above)
Comment (PraveenKumar-NOAA) [reply parm/atm/obs/config/sondes.yaml]: @CoryMartin-NOAA fixed.

File: ush/ufoeval/run_ufo_hofx_test.sh [hunk 10]
```diff
@@ -11,10 +11,10 @@
 #-------------------------------------------------------------
 #--------------- User modified options below -----------------
 cycle=2021080100
-obtype=amsua_n19
-GDASApp=/work2/noaa/da/cmartin/GDASApp/dev/GDASApp # Change this to your own branch
-workdir=/work2/noaa/da/$LOGNAME/ufoeval/$cycle/$obtype
-yamlpath=$GDASApp/parm/atm/obs/testing/amsua_n19.yaml
+obtype=sondes
```
Comment (CoryMartin-NOAA) [inline ush/ufoeval/run_ufo_hofx_test.sh]: Please 'uncommit' this file, otherwise it looks good

File: parm/atm/obs/testing/sondes.yaml (hunk 8 above)
Comment (PraveenKumar-NOAA) [reply parm/atm/obs/testing/sondes.yaml]: @CoryMartin-NOAA done, thanks.

**Review by CoryMartin-NOAA — APPROVED** [review]

File: ush/ufoeval/run_ufo_hofx_test.sh (hunk 10 above)
Comment (PraveenKumar-NOAA) [reply ush/ufoeval/run_ufo_hofx_test.sh]: @CoryMartin-NOAA done, thanks.

File: parm/atm/obs/config/sondes.yaml (hunk 4 above)
Comment (PraveenKumar-NOAA) [reply parm/atm/obs/config/sondes.yaml]: @nicholasesposito please review this PR.

File: parm/atm/obs/testing/sondes.yaml [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+    - name: surface_pressure
+    minvalue: 37499.0
+    maxvalue: 106999.0
+    action:
+      name: reject
+  #
+  # Assign obsError
+  - filter: Perform Action
+    filter variables:
+    - name: surface_pressure
```
Comment (nicholasesposito) [inline parm/atm/obs/testing/sondes.yaml:67]: I see lots of snake style variable names in this file. Just to be clear, this is correct?

Comment (CoryMartin-NOAA) [reply parm/atm/obs/testing/sondes.yaml:67]: for now, yes, we will have to go through GDASApp soon and change these once the code sprint work is in UFO/develop

Comment (nicholasesposito) [reply parm/atm/obs/testing/sondes.yaml:67]: OK

**Review by nicholasesposito — APPROVED** [review]
